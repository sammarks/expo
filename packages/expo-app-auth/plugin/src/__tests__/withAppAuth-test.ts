import { setGradlePlaceholders } from '../withAppAuth';

const buildGradleFixture = `
android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    defaultConfig {
        applicationId "com.helloworld"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName "1.0"
    }
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }

        }
    }
}
`;

describe(setGradlePlaceholders, () => {
  it(`adds placeholders to gradle`, () => {
    expect(setGradlePlaceholders(buildGradleFixture, 'kirkland-brand')).toMatchSnapshot();
  });
});
