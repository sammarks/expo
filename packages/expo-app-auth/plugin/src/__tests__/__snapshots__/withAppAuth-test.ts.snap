// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setGradlePlaceholders adds placeholders to gradle 1`] = `
"
android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    defaultConfig {
        manifestPlaceholders = [appAuthRedirectScheme: 'kirkland-brand']
        applicationId \\"com.helloworld\\"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 1
        versionName \\"1.0\\"
    }
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            def versionCodes = [\\"armeabi-v7a\\": 1, \\"x86\\": 2, \\"arm64-v8a\\": 3, \\"x86_64\\": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }

        }
    }
}
"
`;
